File: core/src/main/java/com/toocol/termio/core/term/commands/processors/HistoryCmdProcessor.java
package com.toocol.termio.core.term.commands.processors;

import com.toocol.termio.core.term.commands.TermCommandProcessor;
import com.toocol.termio.core.term.core.HistoryOutputInfoHelper;
import com.toocol.termio.core.term.core.Term;
import com.toocol.termio.core.term.core.TermStatus;
import com.toocol.termio.utilities.utils.Tuple2;
import io.vertx.core.eventbus.EventBus;

public class HistoryCmdProcessor extends TermCommandProcessor {

      private final HistoryOutputInfoHelper historyOutputInfoHelper = HistoryOutputInfoHelper.getInstance();

      public void process(EventBus eventBus, String cmd, Tuple2<Boolean, String> resultAndMsg) {
          Term.status = TermStatus.HISTORY_OUTPUT;
          historyOutputInfoHelper.displayInformation();
      }

}

File: core/src/main/java/com/toocol/termio/core/term/commands/processors/HistoryCmdProcessor.kt
package com.toocol.termio.core.term.commands.processors

import com.toocol.termio.core.term.commands.TermCommandProcessor
import com.toocol.termio.core.term.core.HistoryOutputInfoHelper.Companion.instance
import com.toocol.termio.core.term.core.Term
import com.toocol.termio.core.term.core.TermStatus
import com.toocol.termio.utilities.utils.Tuple2
import io.vertx.core.eventbus.EventBus

class HistoryCmdProcessor : TermCommandProcessor() {
    private val historyOutputInfoHelper = instance
    override fun process(eventBus: EventBus, cmd: String, resultAndMsg: Tuple2<Boolean, String?>): Any? {
        Term.status = TermStatus.HISTORY_OUTPUT
        historyOutputInfoHelper.displayInformation()
        resultAndMsg.first(true)
        return null
    }
}